client/components/TestPlanReportStatusDialog/index.jsx

```diff
@@ -77,6 +90,55 @@ const TestPlanReportStatusDialog = ({
             return [matched, unmatchedTestPlan, unmatchedRequired];
         }, [testPlanReports, requiredReports]);
 
+    if (newOthers.length > 0) {
+        let index = 0;
+        newOthers.forEach(otherReport => {
+            if (matchedReports.length > 0) {
+                matchedReports.forEach(matchedReport => {
+                    if (
+                        otherReport.at.id === matchedReport.id &&
+                        otherReport.browser.id === matchedReport.browser.id
+                    ) {
+                        delete newOthers[index];
+                    }
+                });
+            }
+            index += 1;
+        });
+
+        index = 0;
+        newOthers.forEach(otherReport => {
+            if (unmatchedTestPlanReports.length > 0) {
+                unmatchedTestPlanReports.forEach(unmatchedTestPlanReport => {
+                    if (
+                        otherReport.at.id === unmatchedTestPlanReport.at.id &&
+                        otherReport.browser.id ===
+                            unmatchedTestPlanReport.browser.id
+                    ) {
+                        delete newOthers[index];
+                    }
+                });
+            }
+            index += 1;
+        });
+
+        index = 0;
+        newOthers.forEach(otherReport => {
+            if (unmatchedRequiredReports.length > 0) {
+                unmatchedRequiredReports.forEach(unmatchedRequiredReport => {
+                    if (
+                        otherReport.at.id === unmatchedRequiredReport.id &&
+                        otherReport.browser.id ===
+                            unmatchedRequiredReport.browser.id
+                    ) {
+                        newOthers.splice(index, 1);
+                    }
+                });
+            }
+            index += 1;
+        });
+    }
```

Comment:
I'm not sure if it's originating in this code, but I noticed that, when I added reports, the order of the reports also changed. It's like the rows with missing reports floated to the top. Personally I feel like the expected behavior would be that the order or reports does not change to avoid disorienting the user. Is that something you think you'd be able to fix?

Reply:
I have a feeling that we can make this code slightly less verbose. Perhaps we can take a look together later.